message/query_options.go

```diff
@@ -322,3 +339,31 @@ func DecodeQueryOptions(source io.Reader, version primitive.ProtocolVersion) (op
 	}
 	return options, nil
 }
+
+func CloneValuesSlice(o []*primitive.Value) []*primitive.Value {
```

Comment:
Do we need this to be public?

Reply:
If so, again I think it would be more elegant to transform this into a `PositionalValues` type with a `Clone()` method.

Reply:
Discussed offline, this involves changing the existing code that encodes and decodes these protocol types so it's out of scope of this PR.

Reply:
OK but they can be private no?

Reply:
done